scrapy/exporters.py

```diff
@@ -128,33 +129,52 @@ def __init__(self, file, **kwargs):
             self.encoding = 'utf-8'
         self.xg = XMLGenerator(file, encoding=self.encoding)
 
+    def _beautify_newline(self):
+        if self.indent_width:
```

Comment:
It seems the meaning on indent_width is different for json and xml: zero value means no newlines for XML, but for json new lines are still inserted, i.e. None and 0 are different for json.